oxidation/libparsec/crates/platform_local_db/src/executor.rs

```diff
@@ -1,178 +1,252 @@
 // Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
 //! Module that wrap an [diesel::SqliteConnection] behind a executor to allow to have an async manner to executor sql queries.
 
-use std::thread::JoinHandle;
-
 use diesel::{connection::SimpleConnection, SqliteConnection};
 use libparsec_platform_async::channel;
+use std::sync::atomic::Ordering;
+use std::sync::{atomic::AtomicBool, Arc, Mutex};
 
 use crate::{DatabaseError, DatabaseResult};
 
-/// The executor that manage and send job to the background executor.
-pub(crate) struct SqliteExecutor {
-    /// The channel that will be used to send job to the background executor.
-    job_sender: Option<channel::Sender<Operation>>,
-    /// The handle to the [BackgroundSqliteExecutor].
-    handle: Option<JoinHandle<()>>,
-}
-
-/// A list of operation that can be executed on the background executor.
-enum Operation {
+enum JobOperation {
     /// Operate a full vacuum, meaning executing `VACUUM` on the sqlite connection
     /// and re-opening the connection to force cleanup of the driver.
-    FullVacuum(channel::Sender<DatabaseResult<()>>),
-    /// Execute a standard job.
-    Job(JobFunc),
+    FullVacuum(JobResultSender<DatabaseResult<()>>),
+    /// Execute an arbitrary operation with the SQLite connection.
+    Fn(JobFunc),
+    /// Dummy operation to wake up the background executor for stop
+    CheckTerminatedFlag,
+}
+
+#[derive(Debug)]
+#[must_use]
+struct JobHandle<R>(channel::Receiver<R>)
+where
+    R: Send + 'static;
+impl<R> JobHandle<R>
+where
+    R: Send + 'static,
+{
+    pub async fn join(self) -> DatabaseResult<R> {
+        self.0.recv_async().await.map_err(|_| DatabaseError::Closed)
+    }
+}
```

Comment:
Not sure what you mean :/

Do you want to do `hande.await` instead of `handle.join().await` ? I'd rather keep the `handle.join().await` to make it more explicit and avoid `submit(...).await.unwrap().await`

Reply:
Keep it that way then